obelisk/cpp/obelisk_cpp/include/obelisk_node.h

```
@@ -0,0 +1,451 @@
+                throw std::runtime_error(
+                    "Provided message type is not a valid Obelisk message!");
+            }
+        } else {
+            RCLCPP_WARN_STREAM(
+                this->get_logger(),
+                "Creating a subscriber that can publish non-Obelisk messages. "
+                "This may cause certain API incompatibilities.");
+        }
+
+        return rclcpp_lifecycle::LifecycleNode::create_subscription<
+            MessageT, CallbackT, AllocatorT, SubscriptionT,
+            MessageMemoryStrategyT>(topic_name, qos, std::move(callback),
+                                    options, msg_mem_strat);
+    };
+
+  protected:
+    /**
+     * @brief Create a subscriber from a configuration string.
+     *
+     * @param config the configuration string
+     * @param callback the callback function (generall of the form
+     * std::bind(&MyFunc, this, _1))
+     * @return the subscription
+     */
+    template <
+        typename MessageT, typename CallbackT,
+        typename AllocatorT    = std::allocator<void>,
+        typename SubscriptionT = rclcpp::Subscription<MessageT, AllocatorT>>
+    std::shared_ptr<SubscriptionT>
+    CreateSubscriptionFromConfigStr(const std::string& config,
+                                    CallbackT&& callback) {
+        // Parse the configuration string
+        const auto config_map                = ParseConfigStr(config);
+        const std::string topic              = GetTopic(config_map);
+        const int depth                      = GetHistoryDepth(config_map);
+        const bool non_obelisk               = !GetIsObeliskMsg(config_map);
+
+        rclcpp::CallbackGroup::SharedPtr cbg = nullptr; // default group
+        try {
+            // Get the callback group based on the string name
+            cbg = callback_groups_.at(GetCallbackGroupName(config_map));
+        } catch (const std::exception& e) {
+        }
```

is it chill style-wise to catch e and do nothing? at least in python they complain about this

---

Modified the code to not have empty catch statements.